packages/gatsby/src/schema/infer-graphql-type.js

```diff
@@ -132,21 +132,31 @@ function inferFromMapping(
   fieldSelector,
   types
 ): ?GraphQLFieldConfig<*, *> {
-  const matchedTypes = types.filter(
-    type => type.name === mapping[fieldSelector].split(`.`)[0]
-  )
+  const linkedType = mapping[fieldSelector].split(`.`)[0]
+  const linkedField =
+    mapping[fieldSelector].slice(linkedType.length + 1) || `id`
+
+  const matchedTypes = types.filter(type => type.name === linkedType)
   if (_.isEmpty(matchedTypes)) {
     console.log(`Couldn't find a matching node type for "${fieldSelector}"`)
     return null
   }
 
   const findNode = (fieldValue, path) => {
-    const linkedType = mapping[fieldSelector].split(`.`)[0]
-    const linkedField = mapping[fieldSelector].split(`.`).slice(1).join(`.`) || `id`
-    const linkedNode = _.find(
+    const linkedNodes = _.filter(
       getNodes(),
-      n => n.internal.type === linkedType && _.get(n, linkedField) === fieldValue
+      n =>
+        n.internal.type === linkedType && _.get(n, linkedField) === fieldValue
     )
+    if (linkedNodes.length > 1) {
+      console.log(
+        `Multiple matching nodes found for "${
+          mapping[fieldSelector]
+        }", returning the first match`
+      )
```

Comment:
This potentially could spam a lot. I think this should log only once for each mapping

Reply:
Yeah, I'm not sure it adds a huge amount of value anyway -- thinking it may make more sense to add a note to docs that only the first match is returned instead.

It doesn't look like `debug` is used very heavily in the code base, but would this maybe be a good use case for it? Not sure if this would actually provide enough info to help people identify why it's matching multiple documents.
```js
const debug = require(`debug`)(`gatsby:schema:infer-graphql-type`)
...
debug(`Multiple matching nodes found for ${fieldSelector} > ${mapping[fieldSelector]}`)
```

Reply:
Yeah, let's just skip this log/warning for now and update docs instead. We can always revisit this later.